PTGLgraphComputation/src/plcc/Main.java

```diff
@@ -5784,9 +5977,273 @@ private static long calculateSkipNeighborNum(Molecule mol1, Molecule mol2, int m
 
     }
 
+    /**
+     * Analyses the statistics of an atom contact and writes the MolContactInfo.
+     * @param x one of the atoms of the residues of the residue pair
+     * @param y one of the atoms of the residues of the residue pair
+     * @param currentMci MolContactInfo of the residue pair of the residues of x and y
+     * @param i index of atom x
+     * @param j index of atom y
+     * @param contactAtomNumInResidueA holds the number Atom x has in its residue a for the contact with minimal distance of that type
+     * @param contactAtomNumInResidueB holds the number Atom y has in its residue a for the contact with minimal distance of that type
+     * @param aIntID internal AA ID of residue a
+     * @param bIntID internal AA ID of residue a
+     * @return A map of atoms of residue b mapped to residue atoms of a.
+     */
+    public static MolContactInfo atomContactsStatistics(Atom x, Atom y, MolContactInfo currentMci, Integer i, Integer j, Integer[] contactAtomNumInResidueA, Integer[] contactAtomNumInResidueB, Integer aIntID, Integer bIntID){
```

Comment:
Basically, we have here a, b, i, j, x and y as variables. I would suggest using more expressive variable names. There is no problem at all when the names are getting longer, for example, atomOfResA or indexOfAtomInResA.